pkg/query/querier.go

```diff
@@ -133,6 +133,16 @@ type querier struct {
 	selectTimeout           time.Duration
 	shardInfo               *storepb.ShardInfo
 	seriesStatsReporter     seriesStatsReporter
+
+	returnChunksMtx sync.Mutex
+	returnChunks    []*storepb.AggrChunk
+}
+
+var returnChunksSlicePool = sync.Pool{
+	New: func() interface{} {
+		r := make([]*storepb.AggrChunk, 0)
+		return &r
+	},
```

Comment:
Maybe we can use the `pool.BucketedPool` and avoid casts down the road and the need to add a new explicit mutex?

Reply:
What min/max sizes and factor would you suggest?

Reply:
Yeah, you got a point as well. Kinda hard to estimate average chunks touched by queries. Let's leave it like this then.